### PR TITLE
Feature/835 brightness increase tracking

### PR DESCRIPTION
**Pull request by JulianHammerl**
* **Tickets addressed:** MAXGNC-835
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

The centerOfBrightness module was updated to compute the total normalized brightness and determine the rolling average of the total brightness. This can be used to track the brightness of the asteroid and make sure it is increasing over time.

The reviewers should be aware that this branch is based off the 833-cob-windowing branch which is currently under review and needs to be merged first (commits 1-10).

## Verification

The unit test was updated and passes.

## Documentation

The documentation was updated accordingly.

## Future work

N/A

### CODE REVIEW

**Review by thibaudteil — APPROVED** [review]

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp]: :large_orange_diamond: I know you didn't name this method, but can we enhance it a little given we're in the neighborhood? Can we give it some verbage like `computeWeighted...` .

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp]: :large_orange_diamond: In a different commit you could remove this `weight` variable and put `auto` in front of the `weight` variable inside the loop.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp]: :large_orange_diamond: It looks like we could get away with adding this variable to hold the state of `brightness` by having  `weightedCenterOfBrightness` call and pass the brightness value to `updateBrightnessHistory`.

Additional state is more logic to have to understand, juggle, and get right. Minimizing state management in our code reduces the possible places for errors/bugs.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp]: :large_orange_diamond: You can remove this line by assigning the output of this call to `mean()` directly to the message output variable.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py:19]: We should rethink this test. But we'll do it in a separate ticket because I think it's complex a job on it's own.

**Review by patkenneally — APPROVED** [review]
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp]: Renamed it to `computeWeightedCenterOfBrightness`.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp]: Done.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp]: Per our chat, I removed the state of `brightness`, but `updateBrightnessHistory` is still called outside of `weightedCenterOfBrightness`. The method `weightedCenterOfBrightness ` now returns both the center of brightness coordinates as well as the total brightness.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp]: Done.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py:19]: Okay.

**Pull request merged**
